### PR TITLE
docs: Update intro documentation page

### PR DESCRIPTION
**Pull request by adchia**
**What this PR does / why we need it**:
This updates the intro Feast documentation page to clarify on:
- What Feast is
- What Feast doesn't fully do
- Who Feast is built for
- Example use cases

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,14 @@
 
 ## Overview
 
-Feast is an open source feature store for machine learning. Feast is the fastest path to productionizing analytic data for model training and online inference.
+Feast (**Fea**ture **St**ore) is an open source feature store for machine learning. Feast is the fastest path to manage existing infrastructure to productionize analytic data for model training and online inference.
+
+
+Feast allows ML platform teams to:
+
+* **Make features consistently available for training and serving** by managing an _offline store_ (to process historical data for scale-out batch scoring or model training), a low-latency _online store_ (to power real-time prediction)_,_ and a battle tested _feature server_ (for serving pre-computed features online).&#x20;
```
Comment (kevjumba) [inline README.md:26]: nit: battle-tested

**Review by kevjumba — APPROVED** [review]

**Review by achals — APPROVED** [review]
/lgtm

File: README.md (hunk 1 above)
Comment (adchia) [reply README.md]: done

**Pull request merged**